src/batou/secrets/tests/test_secrets.py

```diff
@@ -44,163 +44,49 @@ def cleanup_gpg_sockets():
+        with open(cleartext_file) as cleartext:
+            assert cleartext.read().strip() == secret.cleartext.strip()
 
 
 def test_decrypt_missing_key(monkeypatch, encrypted_file):
     monkeypatch.setitem(os.environ, "GNUPGHOME", "/tmp")
 
     with pytest.raises(batou.GPGCallError):
-        EncryptedConfigFile(encrypted_file)
+        with GPGEncryptedFile(encrypted_file) as secret:
+            secret.cleartext
 
 
 def test_write_should_fail_unless_write_locked(encrypted_file):
-    with EncryptedConfigFile(encrypted_file) as secrets:
-        secrets.main_file.cleartext = """\
-[batou]
-members = batou
-[asdf]
-x = 1
-"""
-        secrets.read()
-
-        with pytest.raises(RuntimeError):
-            secrets.write()
-
-
-def test_open_nonexistent_file_for_read_should_fail():
-    with pytest.raises(IOError):
-        EncryptedConfigFile("/no/such/file").__enter__()
-
-
-def test_open_nonexistent_file_for_write_should_create_template_file():
-    tf = tempfile.NamedTemporaryFile(prefix="new_encrypted.")
-    tf.close()  # deletes file
-    encrypted = EncryptedConfigFile(tf.name, write_lock=True)
-    with encrypted as secrets:
-        assert secrets.main_file.cleartext == NEW_FILE_TEMPLATE
-        # The file exists, because we set the write lock
-        assert os.path.exists(secrets.main_file.encrypted_filename)
-    # When saving a config without members, delete the file.
-    assert not os.path.exists(secrets.main_file.encrypted_filename)
-
-
-def test_write_unparsable_raises_error(encrypted_file):
-    encrypted = EncryptedConfigFile(encrypted_file, write_lock=True)
-    with encrypted as secrets:
-        secrets.main_file.cleartext = "some new file contents\n"
-        with pytest.raises(configparser.Error):
-            secrets.read()
-
-
-def test_write(encrypted_file):
-    encrypted = EncryptedConfigFile(encrypted_file, write_lock=True)
-    with encrypted as secrets:
-        secrets.main_file.cleartext = """\
-[batou]
-members = batou
-[asdf]
-x = 1
-"""
-        secrets.read()
-        secrets.write()
-
-    assert FIXTURE_ENCRYPTED_CONFIG.read_bytes() != encrypted_file.read_bytes()
-    assert 0 != encrypted_file.stat().st_size
-
-
-def test_write_fails_without_recipients(encrypted_file):
-    encrypted = EncryptedConfigFile(encrypted_file, write_lock=True)
-    with encrypted as secrets:
-        secrets.main_file.cleartext = """\
-[batou]
-members =
-[asdf]
-x = 1
-"""
-        secrets.read()
-        with pytest.raises(ValueError) as exc:
-            secrets.write()
-
-        assert str(exc.value).startswith("Need at least one recipient.")
-
-
-def test_write_fails_if_recipient_key_is_missing_keeps_old_file(encrypted_file):
-    encrypted = EncryptedConfigFile(encrypted_file, write_lock=True)
-    with encrypted as secrets:
-        secrets.main_file.cleartext = """\
-[batou]
-members = foobar@example.com
-[asdf]
-x = 1
-"""
-        secrets.read()
+    with GPGEncryptedFile(encrypted_file) as secret:
```

Comment:
Please review the missing tests  on the left: most of those are likely paid in blood and I'd like to avoid cutting ourselves again with those historic mistakes again.